src/themes/material/components/chip-list.component.ts

```diff
@@ -0,0 +1,203 @@
+		this.chipValueMap = values.map((itemValue, itemIndex) => {
+			return <IChipValue> {
+				value: itemValue,
+				index: itemIndex
+			};
+		});
+	};
+
+	private commandSet: ICommandSet = {
+		beginEdit: new Command({
+			canExecute: () => this.selectedItem.action === Actions.beginEdit,
+			execute: () => {
+				this.selectedItem.item.value = this.chipValues[this.selectedItem.item.index];
+			},
+			shortcut: 'enter'
+		}),
+
+		endEdit: new Command({
+			canExecute: () => this.selectedItem.action === Actions.endEdit &&
+				this.selectedItem.item.value,
+			execute: () => {
+				this.chipValues[this.selectedItem.item.index] = this.selectedItem.item.value;
+
+				this.chipValueMap[this.selectedItem.item.index].value = this.selectedItem.item.value;
+				this.resetSelectedAction();
+				this.cdRef.detectChanges();
+			},
+			shortcut: 'enter'
+		}),
+
+		addItem: new Command({
+			canExecute: () => this.newItem.action === Actions.add &&
+				this.newItem.item.value && !this.findExistingItem(this.selectedItem.item.value),
+			execute: () => {
+				this.chipValues.push(this.newItem.item.value);
+
+				this.chipValueMap.push({
+					value: this.newItem.item.value,
+					index: this.chipValueMap.length
+				});
+
+				this.resetAddNewAction();
+			},
+			shortcut: 'enter'
+		}),
+
+		deleteItem: new Command({
+			canExecute: () => this.selectedItem.action === Actions.delete,
+			execute: () => {
+				this.chipValues.splice(this.selectedItem.item.index, 1);
+
+				this.chipValueMap = this.chipValues.map((itemValue, itemIndex) => {
+					return <IChipValue> {
+						value: itemValue,
+						index: itemIndex
+					};
+				});
+				this.resetSelectedAction();
+				this.cdRef.detectChanges();
+			}
+		})
+	};
+
+	constructor(private cdRef: ChangeDetectorRef) {
+		const manager = new CommandManager();
+
+		const shortcutCommands = [this.commandSet.endEdit, this.commandSet.addItem];
+		const otherCommands = [this.commandSet.endEdit, this.commandSet.addItem, this.commandSet.deleteItem];
+
+		this.shortcut.register(manager, shortcutCommands);
+	}
+
+	private setSelectedAction(action: Actions, itemIndex?: number) {
+		if (itemIndex >= 0) {
+			this.selectedItem.item.index = itemIndex;
+		}
+		this.selectedItem.action = action;
+	}
+
+	private resetSelectedAction() {
+		this.setSelectedAction(Actions.inactive);
+		this.selectedItem.item.value = null;
+		this.selectedItem.item.index = null;
+	}
+
+	private setAddNewAction() {
+		this.newItem.action = Actions.add;
+	}
+
+	private resetAddNewAction() {
+		this.newItem.action = Actions.inactive;
+		this.newItem.item.value = null;
+		this.newItem.item.index = null;
+	}
+
+	private findExistingItem(item: any) {
+		let found = false;
+
+		this.chipValues.forEach((value) => {
+			if (value === item) {
+				found = true;
+			}
+		});
+		return found;
+	}
+
+	private beginEdit(itemIndex: number) {
+		this.setSelectedAction(Actions.beginEdit, itemIndex);
+		if (this.commandSet.beginEdit.canExecute()) {
```

Comment:
All this method with setAction are looks to heavy, maybe u have some other ideas?